java/vector/src/test/java/org/apache/arrow/vector/TestValueVector.java

```diff
@@ -1394,6 +1394,35 @@ public void testFillEmptiesNotOverfill() {
     }
   }
 
+  @Test
+  public void testSetSafeWithArrowBuf() {
+    final int numValues = BaseFixedWidthVector.INITIAL_VALUE_ALLOCATION * 2;
+
+    try (
+        final VarCharVector fromVector = newVector(VarCharVector.class, EMPTY_SCHEMA_PATH,
+            MinorType.VARCHAR, allocator);
+        final VarCharVector toVector = newVector(VarCharVector.class, EMPTY_SCHEMA_PATH,
+            MinorType.VARCHAR, allocator)) {
+      fromVector.setInitialCapacity(numValues);
+      fromVector.allocateNew();
+      for (int i = 0; i < numValues; ++i) {
+        fromVector.setSafe(i, "hello world".getBytes(), 0, 11);
```

Comment:
it might make the test easier to read if 0 and 11 where constants or had comments next to them.

Reply:
fixed.